rust_src/src/lisp.rs

```diff
@@ -882,6 +882,12 @@ impl LispObject {
         )
     }
 
+    pub fn as_marker_or_error(self) -> LispMarkerRef {
+        self.as_marker().unwrap_or_else(
+            || wrong_type!(Qmarkerp, self),
```

Comment:
I get an error `#[warn(unreachable_code)] on by default`. Is it possible that the macro needs a new scope to work ?